File: core/src/io/github/jerukan/game/ui/gamescreen/menus/BuildInfoMenu.java
package io.github.jerukan.game.ui.gamescreen.menus;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.scenes.scene2d.ui.Label;
import io.github.jerukan.game.ui.ButtonGroup;
import io.github.jerukan.game.ui.gamescreen.buttons.UnitBuildButton;
import io.github.jerukan.util.Assets;

public class BuildInfoMenu extends ButtonGroup {

    private UnitBuildMenu unitMenu;

    private Label unitInfo;

    public BuildInfoMenu(UnitBuildMenu unitMenu) {
        this.unitMenu = unitMenu;
        unitInfo = new Label("", Assets.uiskin, "default");
        unitInfo.setWrap(true);

        //table.setWidth(120);
        table.add(unitInfo).width(160);
        table.setPosition(Gdx.graphics.getWidth() - 200, Gdx.graphics.getHeight() - 100);
    }

    @Override
    public void updateVisibility() {
        UnitBuildButton button = unitMenu.getHoveredButton();
        if(button != null) {
            unitInfo.setText("Unit: " + button.getBaseUnit().name
                    + "\nHealth: " + button.getBaseUnit().baseHealth
                    + "\nSpeed: " + button.getBaseUnit().baseSpeed
                    + "\nAttack: " + button.getBaseUnit().baseAttack
                    + "\nType: " + button.getBaseUnit().type
                    + "\n" + button.getBaseUnit().description);
        }
        else {
            unitInfo.setText("");
        }
    }
}

File: core/src/io/github/jerukan/game/ui/gamescreen/menus/BuildInfoMenu.kt
package io.github.jerukan.game.ui.gamescreen.menus

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.scenes.scene2d.ui.Label
import io.github.jerukan.game.ui.ButtonGroup
import io.github.jerukan.game.ui.gamescreen.buttons.UnitBuildButton
import io.github.jerukan.util.Assets

class BuildInfoMenu(private val unitMenu: UnitBuildMenu) : ButtonGroup() {

    private val unitInfo: Label

    init {
        unitInfo = Label("", Assets.uiskin, "default")
        unitInfo.setWrap(true)

        //table.setWidth(120);
        table.add(unitInfo).width(160f)
        table.setPosition((Gdx.graphics.width - 200).toFloat(), (Gdx.graphics.height - 100).toFloat())
    }

    override fun updateVisibility() {
        val button = unitMenu.hoveredButton
        if (button != null) {
            unitInfo.setText("Unit: " + button.baseUnit.name
                    + "\nHealth: " + button.baseUnit.baseHealth
                    + "\nSpeed: " + button.baseUnit.baseSpeed
                    + "\nAttack: " + button.baseUnit.baseAttack
                    + "\nType: " + button.baseUnit.type
                    + "\n" + button.baseUnit.description)
        } else {
            unitInfo.setText("")
        }
    }
}
